File: sorter_c/bubble_sort.cpp

void bubbleSort(int *list,int size)
{
	for(int i=size-1;i>0;--i)
	{
		for(int j=0;j<i;++j)
		{
			if(list[j]>list[j+1])
			{
				int temp=list[j+1];
				list[j+1]=list[j];
				list[j]=temp;
			}
		}
	}
}

File: sorter_c/bubble_sort.rb

def timeDifMillis(start, finish)
   (finish - start) * 1000.0
end

def bubbleSort list
    for i in (list.size-1).downto(2)
        for j in 0..(i-1)
            if list[j]>list[j+1]
                temp=list[j]
                list[j]=list[j+1]
                list[j+1]=temp
            end
        end
    end
end

list = Array.new

for i in 0...10000
	list.push(rand(100))
end

print list
puts

millis = Time.now
bubbleSort(list)
millis= timeDifMillis(millis,Time.now)

print list
print "millis passed: "
puts millis